cdc/sink/mysql/mysql_syncpoint_store.go

```diff
@@ -209,6 +210,19 @@ func (s *mysqlSyncPointStore) SinkSyncPoint(ctx context.Context,
 		log.Error("sync table: begin Tx fail", zap.Error(err))
 		return cerror.WrapError(cerror.ErrMySQLTxnError, err)
 	}
+
+	// we set `tidb_enable_external_ts_read` = off for each txn,
+	// otherwise TiCDC may not be able to write data to downstream TiDB when
+	// the downstream tidb_enable_external_ts_read = on.
+	if err = s.setTidbEnableExternalTsReadOff(ctx, tx); err != nil {
```

Comment:
The transaction is created before setting the variable, so it's not affected by this modification. We should set the variable before starting the transaction.

However, it's a little hard to do this in go mysql driver (here is a related discussing: https://github.com/go-sql-driver/mysql/issues/208). I'll try to find some way to solve this problem :thinking: .

Reply:
Maybe I could manually get a connection, and use this connection to setup the transaction and variables.

Reply:
I tried `s.db.Conn` and use this connection to set variables and start transaction, but it still doesn't work. I don't know why :facepalm:. Need some investigation.